app/database/queries/breeze_buddy/merchant_entities.py

```
@@ -0,0 +1,342 @@
+# ============================================================================
+
+
+async def check_merchant_id_exists(merchant_id: str) -> bool:
+    """Check if merchant_id already exists in merchants table."""
+    query = "SELECT 1 FROM merchants WHERE merchant_id = $1 LIMIT 1"
+
+    try:
+        async for conn in get_db_connection():
+            row = await conn.fetchrow(query, merchant_id)
+            return row is not None
+        return False  # Fallback if connection not available
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)